Makefile

```diff
@@ -52,18 +60,58 @@ help:
 endif
 
 build-env: # Create a new environment with installed packages
-ifndef name
-	@echo "Error: name is required and must be passed in as a keyword argument."
-	@echo "Usage: make build-env name=<ENV_NAME> py=<PYTHON_VERSION>"
-	@exit 1
-endif
-#	Check if py is set, otherwise use the latest supported version
+#	Validate arguments - exit if unsupported arguments are passed
+	@allowed="type name lfs py include_timestamp"; \
+	for arg in $(filter-out build-env,$(MAKECMDGOALS)) $(MAKEFLAGS); do \
+		case $$arg in \
+			*=*) \
+				arg_name=$${arg%%=*}; \
+				if ! echo " $$allowed " | grep -q " $$arg_name "; then \
+					allowed_list=$$(echo $$allowed | sed 's/ /, /g'); \
+					echo "Error: Invalid argument '$$arg_name'. Allowed arguments are: $$allowed_list" >&2; \
+					exit 1; \
+				fi \
+				;; \
+		esac; \
+	done
+	
+#   Handle arguments and set defaults
+#   type
+	@$(eval type ?= simulation)
+	@$(if $(filter-out simulation artifact,$(type)),$(error Error: 'type' must be either 'simulation' or 'artifact', got '$(type)'))
+#	name
+	@$(eval name ?= $(PACKAGE_NAME)_$(type))
+#	timestamp
+	@$(eval include_timestamp ?= no)
+	@$(if $(filter-out yes no,$(include_timestamp)),$(error Error: 'include_timestamp' must be either 'yes' or 'no', got '$(include_timestamp)'))
+	@$(if $(filter yes,$(include_timestamp)),$(eval override name := $(name)_$(shell date +%Y%m%d_%H%M%S)),)
+#	lfs
+	@$(eval lfs ?= no)
+	@$(if $(filter-out yes no,$(lfs)),$(error Error: 'lfs' must be either 'yes' or 'no', got '$(lfs)'))
+#	python version
 	@$(eval py ?= $(shell python -c "import json; versions = json.load(open('python_versions.json')); print(max(versions, key=lambda x: tuple(map(int, x.split('.')))))"))
+	
 	conda create -n $(name) python=$(py) --yes
 # 	Bootstrap vivarium_build_utils into the new environment
 	conda run -n $(name) pip install vivarium_build_utils
-	conda run -n $(name) make install
+#	Install packages based on type
+	@if [ "$(type)" = "simulation" ]; then \
+		conda run -n $(name) make install ENV_REQS=dev; \
+		conda install -n $(name) redis -c anaconda -y; \
+	elif [ "$(type)" = "artifact" ]; then \
+		conda run -n $(name) make install ENV_REQS=data; \
+	fi
+	@if [ "$(lfs)" = "yes" ]; then \
+		conda run -n $(name) conda install -c conda-forge git-lfs --yes; \
```

Comment:
I assume it doesn't really matter where we get git-lfs? The enviornment.sh didn't install it so users must either never have actually used it or they ust downloaded it on the side.